kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolver.java

```diff
@@ -80,7 +78,43 @@ public ProxyResolutionResult resolveProxyRefs(KafkaProxy proxy, Context<?> conte
         Objects.requireNonNull(proxy);
         Objects.requireNonNull(context);
         Set<VirtualKafkaCluster> virtualKafkaClusters = context.getSecondaryResources(VirtualKafkaCluster.class);
-        return resolve(virtualKafkaClusters, Set.of(proxy), context);
+        if (virtualKafkaClusters.isEmpty()) {
+            return EMPTY_RESOLUTION_RESULT;
+        }
+
+        CommonDependencies commonDependencies = getCommonDependenciesFrom(context);
```

Comment:
it would have been cool if we could call `resolveClusterRefs` directly but from experience I don't think we want to be making multiple calls into the `Context` as the dependencies can change underneath us. Would prefer to have a consistent set of Context deps for the whole resoluation.